### PR TITLE
Add JSDoc for getters

### LINKED ISSUE #6652: HIde undocumented setters and getters from public API
It seems those are currently visible as public API. Example in the CameraComponent:

https://api.playcanvas.com/classes/Engine.CameraComponent.html#renderSceneColorMap

check the code it links .. there is no JSDocs block, but it's still exposed as public on the docs.

Screenshot 2024-06-03 at 15 39 57

### PR DESCRIPTION
**Pull request by willeastcott**
Fixes #6652

Before:
Any undocumented symbols would be ignored in the API reference by TypeDoc _except_ getters, because we only used to document the setter. This meant that some deliberately undocumented getters were making it into the API.

Now:
TypeDoc is now configured to ignore _all_ undocumented symbols. Even getters. To handle this, this PR documents all getters considered to be part of the API.

This PR also improves on and fixes many formatting issues in the getter/setter API. And we no longer have to hack things to get TypeDoc to generate a valid API reference.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/components/screen/component.js [hunk 1]
```diff
@@ -231,13 +242,19 @@ class ScreenComponent extends Component {
         this._elements.forEach(element => element._onScreenSpaceChange());
     }
 
+    /**
+     * Gets whether the ScreenComponent will render its child {@link ElementComponent}s in screen
+     * space instead of world space.
```
Comment (kungfooman) [inline src/framework/components/screen/component.js:247]: I saw you added dashes to "object space" everywhere, should screen space and world space have the same treatment?

Comment (willeastcott) [reply src/framework/components/screen/component.js:247]: I've just done a Google search and it seems more common to not use a hyphen. So I've backed all that out.

Comment (kungfooman) [reply src/framework/components/screen/component.js:247]: Thank you, there still is `object-space` now?

Comment (willeastcott) [reply src/framework/components/screen/component.js:247]: Oh yeah! Forgot about that one. Thanks. Fixed.

File: src/core/shape/oriented-box.js
Comment (LeXXik) [inline src/core/shape/oriented-box.js:1]: Hmm, so if we take `worldTransform` as a property to look at. Do both `_worldTransform` and accessors need to specify the type?

Comment (willeastcott) [reply src/core/shape/oriented-box.js:1]: If you don't specify the type on the `get`, ESLint complains about the JSDoc block having no `@returns` tag. The TypeScript compiler `tsc` can probably infer types...but what we have here works.

Comment (LeXXik) [reply src/core/shape/oriented-box.js:1]: How about removing it from the `_property` then? It seems the whole block is no needed there.

Comment (willeastcott) [reply src/core/shape/oriented-box.js:1]: I think you're right about that. But this PR is already enormous and at least it works. So I would rather get this merged and then any refinement can happen in a follow up PR.

File: src/extras/gizmo/transform-gizmo.js [hunk 2]
```diff
@@ -381,59 +381,79 @@ class TransformGizmo extends Gizmo {
...
+     * Sets whether snapping is enabled. Defaults to false.
      *
      * @type {boolean}
      */
     set snap(value) {
         this._snap = this.root.enabled && value;
     }
 
+    /**
+     * Gets whether snapping is enabled. Defaults to false.
```
Comment (LeXXik) [inline src/extras/gizmo/transform-gizmo.js:393]: ... whether or not .... ?

File: src/core/shape/oriented-box.js
Comment (kungfooman) [reply src/core/shape/oriented-box.js:1]: As long there is still a constructor assignment to infer the types from the type comment block seems superfluous indeed, but if there isn't any constructor assignment we could decrease Intellisense type covering for developers (not sure how many cases that would be overall, probably a few).

File: src/extras/gizmo/transform-gizmo.js (hunk 2 above)
Comment (willeastcott) [reply src/extras/gizmo/transform-gizmo.js:393]: I think that's implicit.

File: src/framework/asset/asset.js [hunk 3]
```diff
@@ -302,14 +312,19 @@ class Asset extends EventHandler {
         }
     }
 
+    /**
+     * Gets the file details or null if no file.
+     *
+     * @type {object}
```
Comment (LeXXik) [inline src/framework/asset/asset.js:318]: {object | null}

File: src/framework/asset/asset.js [hunk 4]
```diff
@@ -357,12 +382,18 @@ class Asset extends EventHandler {
...
+     *
+     * @type {object[]}
+     */
     get resources() {
         return this._resources;
     }
 
     /**
-     * If true the asset will be loaded during the preload phase of application set up.
+     * Sets whether to preload an asset. If true the asset will be loaded during the preload phase
```
Comment (LeXXik) [inline src/framework/asset/asset.js:395]: If true, the ...

Comment (willeastcott) [reply src/framework/asset/asset.js]: Fixed.

File: src/framework/components/collision/component.js [hunk 5]
```diff
@@ -201,12 +224,17 @@ class CollisionComponent extends Component {
...
+     *
+     * @type {Quat}
+     */
     get angularOffset() {
         return this.data.angularOffset;
     }
 
     /**
-     * The radius of the sphere, capsule, cylinder or cone-shaped collision  volumes.
+     * Sets the radius of the sphere, capsule, cylinder or cone-shaped collision  volumes.
```
Comment (LeXXik) [inline src/framework/components/collision/component.js:237]: double space before "volumes"

File: src/framework/components/collision/component.js [hunk 6]
```diff
@@ -215,12 +243,17 @@ class CollisionComponent extends Component {
         this._setValue('radius', arg);
     }
 
+    /**
+     * Gets the radius of the sphere, capsule, cylinder or cone-shaped collision  volumes.
```
Comment (LeXXik) [inline src/framework/components/collision/component.js:247]: Also here

File: src/framework/asset/asset.js (hunk 3 above)
Comment (willeastcott) [reply src/framework/asset/asset.js:318]: There are literally hundreds of these kinds of corrections needed across the codebase. I'd rather leave fixing those out of this PR to simplify things for now (although you might see I have make one or two fixes as I went through documenting the getters). I'd say we just leave them to a follow up PR.

File: src/framework/components/collision/component.js [hunk 7]
```diff
@@ -229,55 +262,76 @@ class CollisionComponent extends Component {
...
+     *
+     * @type {Asset|number|null}
+     */
     get asset() {
         return this.data.asset;
     }
 
     /**
-     * The render asset of the mesh collision volume - can also be  an asset id. Defaults to null.
+     * Sets the render asset or asset id of the mesh collision volume. Defaults to null.
```
Comment (LeXXik) [inline src/framework/components/collision/component.js:314]: "Defaults to null" doesn't seem right on a setter. Should be on a getter probably.

File: src/framework/components/collision/component.js (hunk 5 above)
Comment (willeastcott) [reply src/framework/components/collision/component.js]: Well spotted! Thanks. Fixed.

File: src/framework/components/collision/component.js (hunk 6 above)
Comment (willeastcott) [reply src/framework/components/collision/component.js]: Fixed.

File: src/framework/components/collision/component.js [hunk 8]
```diff
@@ -636,7 +697,8 @@ class CollisionComponent extends Component {
     }
 
     /**
-     * @description Returns the world position for the collision shape taking into account of any offsets.
+     * Returns the world position for the collision shape taking into account of any offsets.
```
Comment (LeXXik) [inline src/framework/components/collision/component.js:701]: ...shape, taking...

File: src/framework/components/collision/component.js [hunk 9]
```diff
@@ -654,7 +716,8 @@ class CollisionComponent extends Component {
     }
 
     /**
-     * @description Returns the world rotation for the collision shape taking into account of any offsets.
+     * Returns the world rotation for the collision shape taking into account of any offsets.
```
Comment (LeXXik) [inline src/framework/components/collision/component.js:720]: shape, taking

Comment (willeastcott) [reply src/framework/components/collision/component.js]: Fixed!

File: src/framework/components/collision/component.js (hunk 8 above)
Comment (willeastcott) [reply src/framework/components/collision/component.js]: Fixed!

File: src/framework/components/element/component.js [hunk 10]
```diff
@@ -796,7 +866,8 @@ class ElementComponent extends Component {
     }
 
     /**
-     * The width of the text rendered by the component. Only works for {@link ELEMENTTYPE_TEXT} types.
+     * Gets the width of the text rendered by the component. Only works for
+     * {@link ELEMENTTYPE_TEXT} types.
```
Comment (LeXXik) [inline src/framework/components/element/component.js:870]: You can safely break links to new line for tighter line fill
```
... for {@link
ELEMENTTYPE_TEXT} types
```

File: src/framework/components/element/component.js [hunk 11]
```diff
@@ -1027,14 +1132,21 @@ class ElementComponent extends Component {
     }
 
     /**
-     * The minimum size that the font can scale to when autoFitWidth or autoFitHeight are true.
+     * Sets the minimum size that the font can scale to when `autoFitWidth` or `autoFitHeight` are
```
Comment (LeXXik) [inline src/framework/components/element/component.js:1135]: good place to use a link {@link autoFitWidth} or {@link autoFitWidth | Auto Fit Width}

File: src/framework/components/gsplat/component.js [hunk 12]
```diff
@@ -69,10 +69,9 @@ class GSplatComponent extends Component {
     }
 
     /**
-     * If set, the object space bounding box is used as a bounding box for visibility culling of
-     * attached gsplat. This allows a custom bounding box to be specified.
+     * Sets a custom object space bounding box for visibility culling of the attached gsplat.
      *
-     * @type {import('../../../core/shape/bounding-box.js').BoundingBox}
+     * @type {import('../../../core/shape/bounding-box.js').BoundingBox|null}
```
Comment (LeXXik) [inline src/framework/components/gsplat/component.js:74]: space before and after |

File: src/framework/components/gsplat/component.js [hunk 13]
```diff
@@ -81,13 +80,19 @@ class GSplatComponent extends Component {
...
+     */
     get customAabb() {
         return this._customAabb;
     }
 
     /**
-     * A {@link GSplatInstance} contained in the component. If not set or loaded, it returns null.
+     * Sets a {@link GSplatInstance} on the component. If not set or loaded, it returns null.
      *
+     * @type {import('../../../scene/gsplat/gsplat-instance.js').GSplatInstance|null}
```
Comment (LeXXik) [inline src/framework/components/gsplat/component.js:95]: here as well

File: src/framework/components/gsplat/component.js [hunk 14]
```diff
@@ -118,6 +123,12 @@ class GSplatComponent extends Component {
         }
     }
 
+    /**
+     * Gets the {@link GSplatInstance} on the component.
+     *
+     * @type {import('../../../scene/gsplat/gsplat-instance.js').GSplatInstance|null}
```
Comment (LeXXik) [inline src/framework/components/gsplat/component.js:129]: and here

Comment (LeXXik) [reply src/framework/components/gsplat/component.js:129]: Hmm, multiple instances. Perhaps a different PR for those.

Comment (mvaligursky) [reply src/framework/components/gsplat/component.js:129]: we have lots of instances of non-spaces around `|` in the engine, I'm not sure we have a strict rule around that at the moment.

File: src/framework/components/element/component.js (hunk 11 above)
Comment (willeastcott) [reply src/framework/components/element/component.js]: Done! I have added lots of other links to this module too.

File: src/framework/components/gsplat/component.js (hunk 14 above)
Comment (willeastcott) [reply src/framework/components/gsplat/component.js:129]: Yeah, we can standardize something in another PR.

File: src/scene/mesh-instance.js [hunk 15]
```diff
@@ -370,12 +370,17 @@ class MeshInstance {
...
+     *
+     * @type {number}
+     */
     get renderStyle() {
         return this._renderStyle;
     }
 
     /**
-     * The graphics mesh being instanced.
+     * Sets the graphics mesh being instanced.
```
Comment (mvaligursky) [inline src/scene/mesh-instance.js:383]: This makes it sounds like mesh instancing is taking place here, I wonder if we could use clearer wording.

**Review by mvaligursky — APPROVED** [review]
Huge effort, even scrolling through this took a while!

File: src/scene/mesh-instance.js (hunk 15 above)
Comment (willeastcott) [reply src/scene/mesh-instance.js:383]: It's the same wording as before, so in the interest in getting this merged, I'll save that for a follow up PR. 😉

**Pull request merged**